client/my-sites/checkout/checkout/index.jsx

```diff
@@ -860,6 +876,10 @@ export class Checkout extends React.Component {
 			analyticsPath = '/checkout/no-site';
 		}
 
+		if ( 'variantShowPlanBump' !== getABTestVariation( 'showPlanUpsellConcierge' ) ) {
+			abtest( 'showPlanUpsellConcierge' );
```

Comment:
> To be clear, recording the ab test event when the checkout page renders will put any user who visits the page into the test, regardless of what's in their cart. Is that right?

That's right, so any user who sees the checkout page is assigned to be part of the test, irrespective of what's in their cart. This does increase noise as you said, but it's a compromise as part of the early bucketing strategy(described in pa1C6h-z7-p2#comment-3282).

The way I understand is that the Hypothesis will be built as AB test start -> purchased Personal plan -> Offer shown (plan bump or concierge) -> Offer purchase. @victorespigares could you confirm that this Hypothesis would work?

I've requested feedback from Data team in pa1C6h-z7-p2#comment-3364.

Reply:
Can't we check if the user has a Personal plan in the cart in the render() bucketing as well? That way we won't include non-eligible users in the test.

Reply:
IMO including non-eligible users in the test does not affect the outcome since we are only interested in the percentage conversion for the steps `purchased Personal plan` -> `Offer shown (plan bump or concierge)` -> `Offer purchase`, and we are not tracking how many people got assigned to abtest_start. Also, even after checking Personal plan in cart we could still have ineligible users in the test if some of those users navigated away from Checkout without purchasing.

I've still gone ahead and [added](https://github.com/Automattic/wp-calypso/pull/36655/commits/bc6165b8af1de9d49f56cced1f810edaf90be7e7) a Personal plan in cart check since it would help save API calls to assign ineligible users.

Reply:
There's been a lot of discussions in pa1C6h-z7-p2, and based on those discussions we are planning to remove early bucketing and launch this test.